lib.coffee

```diff
@@ -308,8 +308,18 @@ class globals.Document
 
       fieldsWalker @meta.fields
 
-      referencesIncludeProjection = LocalCollection._compileProjection referencesInclude
       referencesExcludeProjection = LocalCollection._compileProjection referencesExclude
+      referencesIncludeModification = (document) ->
+        # We can't use _compileProjection here as this would cause top-level fields to
+        # be overwritten when updating subfields.
+        result = {}
+        for path, value of referencesInclude
+          continue if value isnt 1
+
+          value = _.reduce path.split('.'), ((doc, atom) -> doc[atom]), document
+          result[path] = value unless _.isUndefined value
```

Comment:
> I would add here an assert that result[path] does not yet exist. Otherwise it might be overridden.

It can't exist as we are iterating over keys of a dictionary, which are always unique? So no path will be used twice.